server/src/main/java/io/spine/server/entity/storage/Columns.java

```diff
@@ -147,94 +107,158 @@ public static void checkColumnDefinitions(EntityClass<?> entityClass) {
+     *
+     * <p>This way the process of {@linkplain Columns#obtainColumns(Class) obtaining columns} from
+     * the given {@link Entity} class can be skipped.
+     *
+     * @param entity        an {@link Entity} to get the {@linkplain EntityColumn column} values from
+     * @param entityColumns {@linkplain EntityColumn entity columns} which values should be extracted
+     * @param <E>           the type of the {@link Entity}
+     * @return a {@link Map} of the column {@linkplain EntityColumn#getStoredName()
+     *         names for storing} to their {@linkplain MemoizedValue memoized values}
+     * @see MemoizedValue
+     */
+    static <E extends Entity<?, ?>> Map<String, MemoizedValue> extractColumnValues(
+            E entity,
+            Collection<EntityColumn> entityColumns) {
+        checkNotNull(entityColumns);
+        checkNotNull(entity);
+
+        final Class<? extends Entity> entityClass = entity.getClass();
+        if (!isPublic(entityClass)) {
+            return Collections.emptyMap();
         }
-        addToCache(entityType);
+
+        final Map<String, MemoizedValue> values = recordColumnValuesToMap(entityColumns, entity);
+
+        return values;
     }
 
     /**
-     * Caches the {@linkplain Entity entity type}
-     * for further {@linkplain EntityColumn columns} retrieving.
+     * Checks if the given {@link Entity entity class} is public.
+     *
+     * <p>Outputs a message to the log if the class is non-public.
+     *
+     * @param entityClass {@link Entity entity class} to check
+     * @return {@code true} if class is public and {@code false} otherwise
      */
-    private static void addToCache(Class<? extends Entity> entityType) {
-        final BeanInfo entityDescriptor;
-        try {
-            entityDescriptor = Introspector.getBeanInfo(entityType);
-        } catch (IntrospectionException e) {
-            throw new IllegalStateException(e);
+    private static boolean isPublic(Class<? extends Entity> entityClass) {
+        checkNotNull(entityClass);
+
+        final int modifiers = entityClass.getModifiers();
+        if (!Modifier.isPublic(modifiers)) {
+            logNonPublicClass(entityClass);
+            return false;
         }
 
-        final Collection<EntityColumn> entityColumns = newLinkedList();
-        for (PropertyDescriptor property : entityDescriptor.getPropertyDescriptors()) {
-            final Method getter = property.getReadMethod();
-            final boolean isEntityColumn = getAnnotatedVersion(getter).isPresent();
-            if (isEntityColumn) {
-                final EntityColumn column = EntityColumn.from(getter);
-                entityColumns.add(column);
-            }
+        return true;
+    }
+
+    /**
+     * Generates {@linkplain MemoizedValue memoized values} for the given {@linkplain EntityColumn columns}
+     * of the given {@link Entity}.
+     *
+     * <p>Records the result to {@link Map}.
```

Comment:
This is neither here nor there. It's either redundant, since the method signature tells the same. Or, if you wanted to say more about some specifics in the returned `Map` instance (such as describing the keys and values, perhaps, with an example — which I think is redundant for now), then you should've done so.